### PR TITLE
Updates for Julia v0.6

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

**Comment by mlubin** [comment]
Looks like this included some other changes, could you rebase on top of master?

**Comment by blegat** [comment]
Indeed, fixed now :)

File: src/CplexSolverInterface.jl [hunk 1]
```diff
@@ -338,7 +338,8 @@ export cbaddboundbranchup!,
        cbgetdettimestamp,
        cbgetintfeas
 
-abstract CplexCallbackData <: MathProgCallbackData
+using Compat
```
Comment (mlubin) [inline src/CplexSolverInterface.jl]: This should be in ``CPLEX.jl`` near the top.

**Comment by mlubin** [comment]
Testing on 0.5:
```
ERROR: LoadError: LoadError: UndefVarError: @compat not defined
 in include_from_node1(::String) at ./loading.jl:488 (repeats 2 times)
 in macro expansion; at ./none:2 [inlined]
 in anonymous at ./<missing>:?
 in eval(::Module, ::Any) at ./boot.jl:234
 in process_options(::Base.JLOptions) at ./client.jl:242
 in _start() at ./client.jl:321
while loading /home/mlubin/.julia/v0.5/CPLEX/src/cpx_common.jl, in expression starting on line 46
while loading /home/mlubin/.julia/v0.5/CPLEX/src/CPLEX.jl, in expression starting on line 83
```

**Comment by blegat** [comment]
Oups, indeed thanks, fixed in b9e1140

**Comment by mlubin** [comment]
Now:
```
ERROR: LoadError: LoadError: syntax: invalid operator ".!"
 in include_from_node1(::String) at ./loading.jl:488 (repeats 2 times)
 in macro expansion; at ./none:2 [inlined]
 in anonymous at ./<missing>:?
 in eval(::Module, ::Any) at ./boot.jl:234
 in process_options(::Base.JLOptions) at ./client.jl:242
 in _start() at ./client.jl:321
while loading /home/mlubin/.julia/v0.5/CPLEX/src/cpx_vars.jl, in expression starting on line 171
while loading /home/mlubin/.julia/v0.5/CPLEX/src/CPLEX.jl, in expression starting on line 96
ERROR: LoadError: Failed to precompile CPLEX to /home/mlubin/.julia/lib/v0.5/CPLEX.ji.
```

**Comment by blegat** [comment]
Do you know how to find the exact version cutoff like v"0.6.0-dev.1632" for julia PR #17623. How do you find the number "1632" ?

**Comment by mlubin** [comment]
> Do you know how to find the exact version cutoff like v"0.6.0-dev.1632" for julia PR #17623. How do you find the number "1632" ?

Ask @tkelman

**Comment by blegat** [comment]
I have replaced it by an explicit broadcast call.

**Comment by tkelman** [comment]
instructions should be on the compat readme

check out the merge commit then run contrib/commit-name.sh

**Pull request merged**